tmc-langs-framework/src/main/java/fi/helsinki/cs/tmc/langs/io/zip/StudentFileAwareZipper.java

```diff
@@ -110,10 +111,13 @@ private void writeToZip(Path currentPath, ZipArchiveOutputStream zipStream, Path
         log.trace("Writing {} to zip", currentPath);
 
         String name = projectPath.getParent().relativize(currentPath).toString();
+        // zip standard says "/" is to be used as the path separator
+        name = FilenameUtils.separatorsToUnix(name);
```

Comment:
In Unix-based systems it is legal to have `\`-characters in the filename. We don't want to accidentally interpret those as path separators. That's why I would prefer the implementation I proposed earlier.